packages/flutter/test/cupertino/sliding_segmented_control_test.dart

```diff
@@ -2126,4 +2126,30 @@ void main() {
 
     expect(onValueChangedCalled, 0);
   });
+
+  testWidgets('CupertinoSlidingSegmentedControl can be momentary', (WidgetTester tester) async {
```

Comment:
There should also be a test to verify the scale was applied correctly.
Also, looking at all the conditionals around this, are each tested?